WooCommerce/WooCommerceTests/Yosemite/InAppPurchaseStoreTests.swift

```diff
@@ -176,35 +176,43 @@ final class InAppPurchaseStoreTests: XCTestCase {
     }
 
     func test_user_is_entitled_to_product_returns_false_when_not_entitled() throws {
-        // Given
-
         // When
-        let result = waitFor { promise in
-            let action = InAppPurchaseAction.userIsEntitledToProduct(productID: self.sampleProductID) { result in
-                    promise(result)
-                }
-            self.store.onAction(action)
+        Task {
+            do {
+                let isEntitled = try await userIsEntitledToProduct()
+                // Then
+                XCTAssertFalse(isEntitled)
+            } catch {
+                XCTAssert(error is WordPressApiError)
+            }
```

Comment:
If I understand this correctly, the test will pass as long as the error is of type `WordPressApiError`.

But, I am afraid that the original code of this test doesn't expect an error from `userIsEntitledToProduct`. 🤔 

Please correct me if I am missing something.

Reply:
Thanks for pointing this out! I'm trying to retrace the steps to understand why I added this error handling here and honestly I have no idea 🤔 . Removed on https://github.com/woocommerce/woocommerce-ios/pull/8462/commits/04c50065f8f1efc7b81257c7f2b1c2b1002ac528